python/cudf/cudf/tests/test_array_function.py

```diff
@@ -60,7 +60,6 @@ def test_array_func_cudf_dataframe(pd_df, func):
     "func",
     [
         lambda x: np.cov(x, x),
-        lambda x: np.dot(x, x),
```

Comment:
Why was this removed? We still want the dispatch to work, right?

Reply:
I thought this test was assuming that DataFrames don't have `dot()`, and thus is checking for a `TypeError` when we try to call the lambda function on it (at least that's what I gathered from the initial test failures) - is that not the case?

Reply:
Oh you are 100% right, my bad. I just didn't check the body of the test. We should probably add suitable tests for this, then. It looks like previously `np.dot(cudf.Series([1, 2, 3]), cudf.Series([1, 2, 3]))` would work by internal dispatch to `cupy.dot`. Does it now go through `cudf.Frame.dot`, and it works for both `DataFrame` and `Series`? That should get tested now I think.

Reply:
It looks like `np.dot()` is something of a special case for Pandas, in that it returns a Numpy array instead of a DataFrame/Series - is this something we should seek to replicate in cuDF's `__array_function__()`s, or just make a workaround for in the tests (i.e. if we are using `np.dot()` compare the `values` of the result)?

Reply:
I think we should probably aim for the most equivalent approach that also won't destroy performance. In this case, that probably means conversion to a `cupy` array by inserting something a call to the `.values` property in the return statements.